polaris-react/src/components/DataTable/DataTable.tsx

```diff
@@ -468,13 +505,31 @@ class DataTableInner extends PureComponent<CombinedProps, DataTableState> {
       this.setState((prevState) => ({
         ...this.calculateColumnVisibilityData(prevState.condensed),
       }));
-    }, 500);
+    }, 100)();
 
     if (this.props.stickyHeader && this.stickyHeaderActive) {
       this.stickyHeaderScrolling();
     }
   };
```

Comment:
I noticed that debounce like isn't what we want here. Debounce cues up all the calls that would have happened and then eventually fires _all of them_ once the defined delay has passed. There is no reason after 100ms to fire the calculation 50+ times.

I'd suggest to replace the entire handler with the following which gets rid of most of the jank (sorry can't make it a GH suggestion since only the one line is marked as changed):

```tsx
private scrollListener = () => {
    if (this.scrollStopTimer) {
      clearTimeout(this.scrollStopTimer);
    }

    this.scrollStopTimer = setTimeout(() => {
      this.setState((prevState) => ({
        ...this.calculateColumnVisibilityData(prevState.condensed),
      }));
    }, 100);

    this.setState({
      isScrolledFarthestLeft: this.scrollContainer.current?.scrollLeft === 0,
    });

    if (this.props.stickyHeader && this.stickyHeaderActive) {
      this.stickyHeaderScrolling();
    }
};
```

Reply:
This is a huuuge improvement. Thanks for that. Jank is well within acceptable range now.